std/experimental/ndslice/package.d

```diff
@@ -0,0 +1,519 @@
+/+
+## Guide for Slice/BLAS contributors
+
+1. Pay _unprecedented_ attention to functions to be
+       a. inlined(!),
+       b. `@nogc`,
+       c. `nothrow`,
+       d. `pure`.
+    95% of functions will be marked with `pragma(inline, true)`. So, use
+    _simple_ `assert`s with messages that can be computed at compile time.
+    The goals are:
+        1. to reduce executable size for _any_ compilation mode
+        2. to reduce template bloat in object files
+        3. to reduce compilation time
+        4. to allow a user to write an extern C bindings for code based on `Slice`.
+
+2. Do not import `std.format`, `std.string` and `std.conv` to format error
+    messages.`"Use" ~ Concatenation.stringof`.
+
+3. Try to use already defined `mixin template`s for pretty error messaging.
+
+4. Do not use `Exception`s/`enforce`s to check indexes and length. Exceptions are
+    allowed only for algorithms where validation of an input data is
+    significantly complex for user. `reshape` is a good example where
+    Exceptions are required. Put an example of Exception handing and workaround
+    for a function that can throw.
+
+5. Do not use compile time flags for simple checks like transposition
+    of a matrix. It is much better to have runtime matrix transposition.
+    Furthermore, Slice provide runtime matrix transposition out of the box.
+
+6. Do not use _Fortran_VS_C_ flags. They are about notation,
+    but not about algorithm itself. To care about math world users add
+    appropriate code example in the documentation. `transposed` / `everted`
+    can be used for cash friendly code.
+
+7. Do not use D compile time power to produce dummy entities like
+    `IdentityMatrix`.
+
+8. Try to separate allocation and algorithm logic whenever possible.
+
+9. Add CTFE unittests to new functions.
++/
+
+/**
+The `ndslice` package offers a compact and flexible API for working with
+multidimensional arrays and iterators. The package is designed for creating
+machine learning and image processing algorithms as well as for carrying out
+calculations in physics, statistics and linear algebra.
+
```

Comment:
I suggest:

The package is designed for applications such as linear algebra, physics and statistics. It would be well suited to creating machine learning and image processing algorithms, but should also be general enough for use anywhere with homogeneously-typed multidimensional data.